autoscale/src/main/java/com/sequenceiq/periscope/controller/DistroXAutoScaleClusterV1Controller.java

```diff
@@ -135,25 +135,29 @@ private DistroXAutoscaleClusterResponse updateClusterAutoScaleState(Cluster clus
 
     private DistroXAutoscaleClusterResponse updateClusterAutoScaleConfig(Cluster cluster,
             DistroXAutoscaleClusterRequest autoscaleClusterRequest) {
-
-        alertValidator.validateEntitlementAndDisableIfNotEntitled(cluster);
-        alertValidator.validateDistroXAutoscaleClusterRequest(cluster, autoscaleClusterRequest);
-
         if (Boolean.TRUE.equals(autoscaleClusterRequest.getUseStopStartMechanism())) {
             alertValidator.validateStopStartEntitlementAndDisableIfNotEntitled(cluster);
         }
+
+        Boolean enableStopStartScaling = autoscaleClusterRequest.getUseStopStartMechanism();
+        if (enableStopStartScaling == null && autoscaleClusterRequest.getTimeAlertRequests().isEmpty()) {
+            enableStopStartScaling = asClusterCommonService.shouldEnableStopStartBasedOnEntitlement(cluster);
+        }
```

Comment:
Can we move this logic to a separate method (either here or elsewhere) for better readability?

Reply:
Done